packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx

```diff
@@ -163,7 +169,7 @@ export class ColorPickerBase extends BaseComponent<IColorPickerProps, IColorPick
     }
 
     let newColor: IColor | undefined;
-    if (isHex) {
+    if (isHex && newValue && newValue.length === 6) {
```

Comment:
hmm. Is this right? can't this be 3 digits too?

Reply:
The onBlur handles 3 digit hex values. If we were to handle them here, the 3 digit hex value will update to full 6 digit while the user is typing. As in lets say I want to enter "fafddd", as soon as I have entered "faf" it will update it to "ffaaff" and not let me enter the desired value.

Basically, onBlur is handing all updates and onChange is limited to updating only 6 digit values.